tests/common.go

```diff
@@ -815,18 +827,45 @@ func ValidatePureVolumeStatisticsDynamicUpdate(ctx *scheduler.Context, errChan .
 func ValidatePureVolumeSnapshotAndRestore(ctx *scheduler.Context, errChan ...*chan error) {
 	context("For validation of an snapshot and restoring", func() {
 		timestamp := strconv.Itoa(int(time.Now().Unix()))
-		snapShotClassName := PureSnapShotClass + "-" + timestamp
+		snapShotClassName := PureSnapShotClass + "." + timestamp
 		if _, err := Inst().S.CreateCsiSanpshotClass(snapShotClassName, "Delete"); err != nil {
 			logrus.Errorf("Create volume snapshot class failed with error: [%v]", err)
 			expect(err).NotTo(haveOccurred(), "failed to create snapshot class")
 		}
 
-		err := Inst().S.CreateCsiSnapsAndValidate(ctx, snapShotClassName)
+		request := scheduler.CSISnapshotRequest {
+			Timestamp: timestamp,
+			Namespace: "fada-basic-snapshot-and-restore-test-",
+			OriginalPVCName: "fada-basic-pvc",
+			SnapName: "fada-basic-pvc-snapshot",
+			RestoredPVCName: "fada-basic-pvc-restored",
+			StorageclassName: "elasticsearch-sc",
+			SnapshotclassName: snapShotClassName,
+			FADA: true,
+		}
+
+		err := Inst().S.CreateCsiSnapsAndValidate(ctx, request)
 		processError(err, errChan...)
 
 	})
 }
 
+// ValidatePureVolumeClone is the ginkgo spec for cloning a volume and verifying the content
+func ValidatePureVolumeClone(ctx *scheduler.Context, errChan ...*chan error) {
```

Comment:
Can we remove the PureVolume terms to avoid confusion. What you are doing here is majorly doing CSI function (clone and snapshot) validation. what storage we are using doesn't matter.